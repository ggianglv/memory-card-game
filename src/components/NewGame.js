import React, { useContext } from "react";
import { SET_MODE } from "../reducers/actions";
import { LEVEL } from "../constants";
import { AppContext } from "../context";

const NewGame = ({ closeModal }) => {
  const { dispatch } = useContext(AppContext);

  const setMode = (mode) => {
    dispatch({
      type: SET_MODE,
      payload: mode,
    });

    closeModal();
  };

  return (
    <div className="new-game">
      <div onClick={() => setMode(LEVEL.EASY)} className="modal-item">
        Easy (4x4)
      </div>
      <div onClick={() => setMode(LEVEL.MEDIUM)} className="modal-item">
        Medium (4x5)
      </div>
      <div onClick={() => setMode(LEVEL.HARD)} className="modal-item">
        Hard (4x6)
      </div>
    </div>
  );
};

export default NewGame;
