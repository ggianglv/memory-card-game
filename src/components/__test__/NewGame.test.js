import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import NewGame from "../NewGame";
import renderer from "react-test-renderer";

test("render NewGame without crash", () => {
  render(<NewGame />);
});

test("render correct NewGame", () => {
  const { getByTestId } = render(<NewGame />);

  expect(getByTestId("easy")).toHaveTextContent("Easy (4x4)");
  expect(getByTestId("medium")).toHaveTextContent("Medium (4x5)");
  expect(getByTestId("hard")).toHaveTextContent("Hard (4x6)");
});

test("NewGame to match snapshot", () => {
  const tree = renderer.create(<NewGame />).toJSON();

  expect(tree).toMatchSnapshot();
});
