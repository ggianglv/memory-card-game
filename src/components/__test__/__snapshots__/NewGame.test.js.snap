// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewGame to match snapshot 1`] = `
<div
  className="new-game"
>
  <div
    className="modal-item"
    data-testid="easy"
    onClick={[Function]}
  >
    Easy (4x4)
  </div>
  <div
    className="modal-item"
    data-testid="medium"
    onClick={[Function]}
  >
    Medium (4x5)
  </div>
  <div
    className="modal-item"
    data-testid="hard"
    onClick={[Function]}
  >
    Hard (4x6)
  </div>
</div>
`;
